### PR TITLE
REFRESH tags

### LINKED ISSUE #49: Assing Tags to Spins CRUD
A user should be able to create new taggins (associations between a  Tag and a Spin):

- [ ] As a user, I  should be able to assign a tag to his spins, by name. If the tag does not exist, it will be created
- [ ] As a user, I won't be able to assign a tag to a spin that is not mine
- [ ] As a user, I want to be able to update a tag. Updating a tag will imply a new tagging, and possibly the creation of a new tag. Updating is only allowed for own tags.
- [ ] As a user, I want to be able to delete a tag from a spin. Deleting a tag will delete the tagging.
- [ ] As a user, I should be able to show tags for one spin

### PR DESCRIPTION
**Pull request by aljesusg**
Now we have the tags provided by metadata file.
After save the Tag we validate the tag (app/models/tag.rb) with regular expressions provided in file produc/validate_tags.yaml

## Sample
if we have the tag semo, zemo (regular expression matched .emo) this throw the log 'Maybe the tag semo is wrong. Did you mean demo?. '
```yaml
demo: ['.emo']
open-source: ['.pen-source'] 
```

FIX #49 ?

### CODE REVIEW

File: app/models/spin.rb [hunk 1]
```diff
@@ -170,4 +170,14 @@ def validate_metadata?
     end
     false
   end
+
+  def refresh_tags
+    tags.delete_all
+    metadata['tags'].each do |tag|
+      new_tag = Tag.exists?(name: tag) ? Tag.find_by_name(tag) : Tag.create(name: tag)
```
Comment (chargio) [inline app/models/spin.rb]: Tag.Tag.find_or_create

**Comment by chargio** [comment]
@aljesusg Some conflicts with the base branch

**Comment by aljesusg** [comment]
@chargio rdy for merge plz

File: app/models/spin.rb [hunk 2]
```diff
@@ -193,4 +193,16 @@ def update_releases
     update(releases: releases)
     true
   end
+
+  # Refresh tags of spin
+  #
+  def refresh_tags
+    tags.delete_all
```
Comment (chargio) [inline app/models/spin.rb]: 1 user a has tag with namn "uno"
2 user b has also tag with name "uno" in on their spins

3. User b refresh
Tag is deleted --> Taggins are deleted --> Spins from a are no longer tagged

Comment (aljesusg) [reply app/models/spin.rb]: No !!!  this delete only relations

Comment (chargio) [reply app/models/spin.rb]: OK: I didn't realize is delete_all and not destroy_all

**Pull request merged**